sdk/dns/Azure.ResourceManager.Dns/src/Customization/ArmDnsModelFactory.cs

```diff
@@ -358,5 +358,75 @@ public static DnsTxtRecordData DnsTxtRecordData(ResourceIdentifier id = null, st
+        /// <param name="soaRecordInfo"> The SOA record in the record set. </param>
+        /// <param name="caaRecords"> The list of CAA records in the record set. </param>
+        /// <param name="dsRecords"> The list of DS records in the record set. </param>
+        /// <param name="tlsaRecords"> The list of TLSA records in the record set. </param>
+        /// <param name="naptrRecords"> The list of NAPTR records in the record set. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        public static DnsRecordData DnsRecordData(ResourceIdentifier id, string name, ResourceType resourceType, ResourceManager.Models.SystemData systemData, ETag? etag, IDictionary<string, string> metadata, long? ttl, string fqdn, string provisioningState, WritableSubResource targetResource, WritableSubResource trafficManagementProfile, IList<DnsARecordInfo> aRecords, IList<DnsAaaaRecordInfo> aaaaRecords, IList<DnsMXRecordInfo> mxRecords, IList<DnsNSRecordInfo> nsRecords, IList<DnsPtrRecordInfo> ptrRecords, IList<DnsSrvRecordInfo> srvRecords, IList<DnsTxtRecordInfo> txtRecords, string cname, DnsSoaRecordInfo soaRecordInfo, IList<DnsCaaRecordInfo> caaRecords, IList<DnsDSRecordInfo> dsRecords, IList<DnsTlsaRecordInfo> tlsaRecords, IList<DnsNaptrRecordInfo> naptrRecords, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            metadata ??= new Dictionary<string, string>();
+            aRecords ??= new List<DnsARecordInfo>();
+            aaaaRecords ??= new List<DnsAaaaRecordInfo>();
+            mxRecords ??= new List<DnsMXRecordInfo>();
+            nsRecords ??= new List<DnsNSRecordInfo>();
+            ptrRecords ??= new List<DnsPtrRecordInfo>();
+            srvRecords ??= new List<DnsSrvRecordInfo>();
+            txtRecords ??= new List<DnsTxtRecordInfo>();
+            caaRecords ??= new List<DnsCaaRecordInfo>();
+            dsRecords ??= new List<DnsDSRecordInfo>();
+            tlsaRecords ??= new List<DnsTlsaRecordInfo>();
+            naptrRecords ??= new List<DnsNaptrRecordInfo>();
+            serializedAdditionalRawData ??= new Dictionary<string, BinaryData>();
+
+            return new DnsRecordData(
+                id,
+                name: name,
+                resourceType: resourceType,
+                systemData: systemData,
+                etag: etag,
+                metadata: metadata,
+                ttl: ttl,
+                fqdn: fqdn,
+                provisioningState: provisioningState,
+                targetResource: targetResource,
+                trafficManagementProfile: trafficManagementProfile,
+                aRecords: aRecords,
+                aaaaRecords: aaaaRecords,
+                mxRecords: mxRecords,
+                nsRecords: nsRecords,
+                ptrRecords: ptrRecords,
+                srvRecords: srvRecords,
+                txtRecords: txtRecords,
+                cnameRecordInfo: null,
```

Comment:
should this be a new instance with cname?

Reply:
Fixed